classes/local/store/s3/client.php

```diff
@@ -170,7 +170,9 @@ public function get_trash_fullpath_from_hash($contenthash) {
      * @path   string full path to S3 file.
      */
     public function delete_file($fullpath) {
-        unlink($fullpath);
+        $relativepath = $this->get_relative_path_from_fullpath($fullpath);
+
+        $this->client->deleteObject(array('Bucket' => $this->bucket, 'Key' => $relativepath));
```

Comment:
This looks ok at face value, but can you explain why this is better that the stream wrapper unlink which does basically exactly the same thing?

https://github.com/catalyst/moodle-local_aws/blob/master/sdk/Aws/S3/StreamWrapper.php#L216-L225